### ISSUE TITLE
multi-gpu training got stuck

### ISSUE THREAD

**Issue by jiahuigeng** [facebookresearch/XLM#80]
INFO - 05/15/19 14:07:09 - 0:00:04 - ============ Data summary
INFO - 05/15/19 14:07:09 - 0:00:04 - Monolingual data   - train -           en:   5000000
INFO - 05/15/19 14:07:09 - 0:00:04 - Monolingual data   - valid -           en:      3000
INFO - 05/15/19 14:07:09 - 0:00:04 - Monolingual data   -  test -           en:      3003
INFO - 05/15/19 14:07:09 - 0:00:04 - Monolingual data   - train -           fr:   5000000
INFO - 05/15/19 14:07:09 - 0:00:04 - Monolingual data   - valid -           fr:      3000
INFO - 05/15/19 14:07:09 - 0:00:04 - Monolingual data   -  test -           fr:      3003

INFO - 05/15/19 14:07:11 - 0:00:06 - Number of parameters (model): 141848203
INFO - 05/15/19 14:07:11 - 0:00:06 - Number of parameters (model): 141848203
INFO - 05/15/19 14:07:12 - 0:00:07 - Number of parameters (model): 141848203
INFO - 05/15/19 14:07:12 - 0:00:07 - Number of parameters (model): 141848203
INFO - 05/15/19 14:07:14 - 0:00:09 - Using nn.parallel.DistributedDataParallel ...
INFO - 05/15/19 14:07:14 - 0:00:09 - Using nn.parallel.DistributedDataParallel ...
INFO - 05/15/19 14:07:14 - 0:00:09 - Using nn.parallel.DistributedDataParallel ...
INFO - 05/15/19 14:07:14 - 0:00:09 - Using nn.parallel.DistributedDataParallel ...
INFO - 05/15/19 14:07:15 - 0:00:10 - ============ Starting epoch 0 ... ============
INFO - 05/15/19 14:07:15 - 0:00:10 - Creating new training data iterator (pred,fr) ...
INFO - 05/15/19 14:07:15 - 0:00:10 - ============ Starting epoch 0 ... ============
INFO - 05/15/19 14:07:15 - 0:00:10 - Creating new training data iterator (pred,en) ...
INFO - 05/15/19 14:07:15 - 0:00:10 - ============ Starting epoch 0 ... ============
INFO - 05/15/19 14:07:15 - 0:00:10 - Creating new training data iterator (pred,en) ...
INFO - 05/15/19 14:07:15 - 0:00:10 - ============ Starting epoch 0 ... ============
INFO - 05/15/19 14:07:15 - 0:00:10 - Creating new training data iterator (pred,en) ...

When I run the pretrain experiment on a K80 with 4 gpus, it got stuck like that above. Things were different when I ran the same command on V100 machine. I feel strange because no errors and warnings, but stopped at this step for one night.  any hints please?

**Comment by glample** [comment]
Mm when you run with just 1 GPU on the K80 does it work? Can you see if it still gets stuck when you use a very small batch size?

**Comment by glample** [comment]
Closing for inactivity. Fell free to reopen.

**Comment by colmantse** [comment]
I actually have a similar issue.

my setting is 8 v100 gpu

the end of the log before it freezes is:
`INFO - 01/30/21 18:21:43 - 0:00:39 - Number of parameters (model): 64889361
INFO - 01/30/21 18:21:44 - 0:00:40 - Found 0 memories.
INFO - 01/30/21 18:21:44 - 0:00:40 - Found 6 FFN.
INFO - 01/30/21 18:21:44 - 0:00:40 - Found 102 parameters in model.
INFO - 01/30/21 18:21:44 - 0:00:40 - Using nn.parallel.DistributedDataParallel ...
INFO - 01/30/21 18:21:44 - 0:00:40 - Found 0 memories.
INFO - 01/30/21 18:21:44 - 0:00:40 - Found 6 FFN.
INFO - 01/30/21 18:21:44 - 0:00:40 - Found 0 memories.
INFO - 01/30/21 18:21:44 - 0:00:40 - Found 6 FFN.
INFO - 01/30/21 18:21:44 - 0:00:40 - Found 102 parameters in model.
INFO - 01/30/21 18:21:44 - 0:00:40 - Found 102 parameters in model.
INFO - 01/30/21 18:21:44 - 0:00:40 - Using nn.parallel.DistributedDataParallel ...
INFO - 01/30/21 18:21:44 - 0:00:40 - Using nn.parallel.DistributedDataParallel ...
INFO - 01/30/21 18:21:44 - 0:00:40 - Found 0 memories.
INFO - 01/30/21 18:21:44 - 0:00:40 - Found 6 FFN.
INFO - 01/30/21 18:21:44 - 0:00:40 - Found 0 memories.
INFO - 01/30/21 18:21:44 - 0:00:40 - Found 6 FFN.
INFO - 01/30/21 18:21:44 - 0:00:40 - Found 0 memories.
INFO - 01/30/21 18:21:44 - 0:00:40 - Found 6 FFN.
INFO - 01/30/21 18:21:44 - 0:00:40 - Found 102 parameters in model.
INFO - 01/30/21 18:21:44 - 0:00:40 - Using nn.parallel.DistributedDataParallel ...
INFO - 01/30/21 18:21:44 - 0:00:40 - Found 0 memories.
INFO - 01/30/21 18:21:44 - 0:00:40 - Found 6 FFN.
INFO - 01/30/21 18:21:44 - 0:00:40 - Found 102 parameters in model.
INFO - 01/30/21 18:21:44 - 0:00:40 - Using nn.parallel.DistributedDataParallel ...
INFO - 01/30/21 18:21:44 - 0:00:40 - Found 102 parameters in model.
INFO - 01/30/21 18:21:44 - 0:00:40 - Using nn.parallel.DistributedDataParallel ...
INFO - 01/30/21 18:21:44 - 0:00:40 - Found 102 parameters in model.
INFO - 01/30/21 18:21:44 - 0:00:40 - Using nn.parallel.DistributedDataParallel ...

`
however, nvidia-smi shows that 7 of the 8 GPU has usage of 100% while 1 is 0 %
also, top says 7 python processes has cpu usage of 100%

I am not sure what to do because it doesnt log "starting epoch 0" and just remain silent as is. 
also instead of all 8 gpu being used, only 7 is running.

below is the command i used to start the training:
`export NGPU=8; python -m torch.distributed.launch --nproc_per_node=$NGPU train.py --exp_name test_enzh_mlm --dump_path ./dumped/ --data_path ./data/processed/en-zh/ --lgs 'en-zh' --clm_steps 'en,zh' --mlm_steps 'en,zh' --emb_dim 512 --n_layers 6 --n_heads 8 --dropout 0.1 --attention_dropout 0.1 --batch_size 16 --bptt 256 --optimizer adam,lr=0.0001 --epoch_size 200000 --validation_metrics _valid_mlm_ppl --stopping_criterion _valid_mlm_ppl,10
`
